src/features/hotspots/settings/updateHotspot/UpdateHotspotConfig.tsx

```diff
@@ -183,8 +190,35 @@ const UpdateHotspotConfig = ({ onClose, onCloseSettings, hotspot }: Props) => {
     })
   }
 
+  const hasPendingTransaction = useCallback(async () => {
+    try {
+      const pending = (await dispatch(
+        fetchTxnsHead({ filter: 'pending' }),
+      )) as {
+        payload?: HttpTransaction[]
+      }
+      const txns = pending.payload
+      return txns?.find((pendingTxn) => {
+        if (!isPendingTransaction(pendingTxn)) return
+
+        return (
+          pendingTxn.txn.type === 'assert_location_v2' &&
+          pendingTxn.status === 'pending' &&
+          pendingTxn.txn.gateway === hotspot?.address
+        )
+      })
+    } catch (e) {}
+    return false
+  }, [dispatch, hotspot?.address])
+
   const onSubmit = async () => {
     setLoading(true)
+    const hasExistingPendingTxn = await hasPendingTransaction()
```

Comment:
should we do this on mount instead? That way if the API is slow it has more time to load.

Reply:
I feel like it would be weird to have it just pop up though. I think the flow is better so that it can wait until the last moment to check in case the other one clears.

Reply:
It has a nice loading animation in the button, so you at least know its loading